code/modules/food/cooking/cooking_vessels/_cooking_vessel.dm

```diff
@@ -150,3 +150,4 @@
 	. = ..()
 	if(. != PROCESS_KILL && !is_processing)
 		START_PROCESSING(SSobj, src)
+	update_icon()
```

Comment:
this feels kind of gross given that it'll be an icon update every 2 seconds, but cooking vessels are relatively light in terms of icon stuff compared to humantype mobs and similar, so i figure it's probably fine. would still like if there was a way to more intelligently update stuff re: state transitions

actually, with nata's PR, will solid reagents be moved into the liquid list when they reach their melting point? and when water starts boiling it'll start to vaporize and cause a reagent change which will cause an icon update. are you certain this update_icon() call is still necessary?